pkg/skaffold/schema/validation/validation.go

```diff
@@ -219,6 +219,72 @@ func validateUniqueDependencyAliases(artifacts []*latest.Artifact) (errs []error
 	return
 }
 
+// extractContainerNameFromNetworkMode returns the container name even if it comes from an Env Var. Error if the mode isn't valid
+// (only container:<id|name> format allowed)
+func extractContainerNameFromNetworkMode(mode string) (string, error) {
+	if strings.HasPrefix(strings.ToLower(mode), "container:") {
+		// Up to this point, we know that we can strip until the colon symbol and keep the second part
+		// this is helpful in case someone sends container not in lowercase
+		maybeID := strings.SplitN(mode, ":", 2)[1]
+		id, err := util.ExpandEnvTemplate(maybeID, map[string]string{})
+		if err != nil {
+			return "", sErrors.NewError(err,
+				proto.ActionableErr{
+					Message: fmt.Sprintf("unable to parse container name %s: %s", mode, err),
+					ErrCode: proto.StatusCode_TEST_CUSTOM_CMD_PARSE_ERR,
+					Suggestions: []*proto.Suggestion{
+						{
+							SuggestionCode: proto.SuggestionCode_CHECK_CUSTOM_COMMAND,
+							Action:         fmt.Sprintf("Check the content of the environment variable: %s", mode),
+						},
+					},
+				})
+		}
+		return id, nil
+	}
+	errMsg := fmt.Sprintf("extracting container name from a non valid container network mode '%s'", mode)
+	return "", sErrors.NewError(fmt.Errorf(errMsg),
+		proto.ActionableErr{
+			Message: errMsg,
+			ErrCode: proto.StatusCode_TEST_CUSTOM_CMD_PARSE_ERR,
```

Comment:
Reuse the invalid container name error code and suggestion or similar to above comment.

Reply:
Same here